refactor pv e2e code to improve readability

jeffvance opened the pull request:
**What this PR does / why we need it**:
Moved the helper functions out of _persistent_volumes.go_ to a new file, _pvutil.go_, in order to improve readability and make it easier to add new tests.
Also, all pod delete code now calls the same helper function `deletePodWithWait`.

**Release note**:
```
NONE
```

jeffvance committed:
refactor pv code for readability

k8s-reviewable commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.kubernetes.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/40530)
<!-- Reviewable:end -->

jeffvance commented:
@copejon @jsafrane PTAL. This should be quick...

spxtr commented:
@k8s-bot bazel test this
#40526

liggitt commented:
is it a straight move? hard to tel

jeffvance commented:
@liggitt Yes straight move with one caveat: I consolidated use of a common func to delete-a-pod-and-wait-for-it-to-terminate. So, _kubelet.go_'s and _persistent_volumes-disruptive.go_ 's version of deleting a pod has been consolidated into _pvutil.go_. Since the namespace arg is not needed now, `ns` was removed from the calls to deletePod in _persistent_volumes.go_ .
moved the `deletePodwithWait` func from _kubelet.go_ to _pvutil.go_ and changed the code in _persistent_volumes.go_ and in _persistent_volumes-disruptive.go_ to call it rather than using their own, identical func to delete a pod.

liggitt commented on test/e2e/persistent_volumes-disruptive.go:
```diff
@@ -222,7 +222,7 @@ func initTestCase(f *framework.Framework, c clientset.Interface, pvConfig persis
 
 // tearDownTestCase destroy resources created by initTestCase.
 func tearDownTestCase(c clientset.Interface, f *framework.Framework, ns string, pod *v1.Pod, pvc *v1.PersistentVolumeClaim, pv *v1.PersistentVolume) {
-	deletePod(f, c, ns, pod)
```
is namespace always set in pod and identical to ns?

jeffvance replied:
Yes. The pod's ns is the same as used by the test and is set in the pod -- see `makePod()`

liggitt commented on test/e2e/pvutil.go:
```diff
@@ -0,0 +1,597 @@
+	if pod == nil {
+		return
+	}
+	framework.Logf("Deleting pod %v", pod.Name)
```
the old method no-oped on pods with no name

jeffvance replied:
Correct. However. the pods always have a generated name.

jeffvance replied:
@liggitt @saad-ali is there a label that can be applied to re-run the build/test? I looked at the 3 failures below and, unless I'm missing something, they appear to be build/test infra related. Thanks.

liggitt commented:
the "Rerun command" in the bot comment is what you want

liggitt commented:
@k8s-bot unit test this
@k8s-bot verify test this
@k8s-bot bazel test this

jeffvance commented:
@k8s-bot cvm gce e2e test this

jeffvance commented:
@liggitt all tests pass so can I get a /lgtm so we can get this merged? This pr will cause at least one pr in the queue to need to be rebased but that is expected. Thanks.

jeffvance commented:
Rebased.

liggitt commented:
/lgtm

jeffvance commented:
@liggitt Thanks for the lgtm. Can you become an assignee?  Thanks again!

liggitt commented:
/lgtm

deads2k commented:
/approve

jsafrane commented:
@k8s-bot unit test this
